app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java

```diff
@@ -592,6 +611,30 @@ else if (d2.trackedEntityModule().trackedEntityAttributes().byUid().eq(tetAttrib
         return teis;
     }
 
+    private List<TrackedEntityInstance> filterByStatus(List<TrackedEntityInstance> teis, List<EventStatus> eventStatuses) {
+        Iterator<TrackedEntityInstance> iterator = teis.iterator();
+        if (!eventStatuses.isEmpty())
+            while (iterator.hasNext()) {
+                TrackedEntityInstance tei = iterator.next();
+
+                List<Event> eventsToOverdue = d2.eventModule().events().byTrackedEntityInstanceUids(Collections.singletonList(tei.uid())).byStatus().eq(EventStatus.SCHEDULE)
+                        .byDueDate().before(new Date()).blockingGet();
+                for(Event event : eventsToOverdue){
+                    try {
+                        d2.eventModule().events().uid(event.uid()).setStatus(EventStatus.OVERDUE);
+                    } catch (D2Error d2Error) {
+                        d2Error.printStackTrace();
+                    }
+                }
+
+                boolean hasEventWithStatus = !d2.eventModule().events().byTrackedEntityInstanceUids(Collections.singletonList(tei.uid())).byStatus().in(eventStatuses).blockingIsEmpty();
+                if (!hasEventWithStatus)
+                    iterator.remove();
```

Comment:
I don't think this pass the code style from lint :dancer:

Reply:
It's java :P